cmd/firectl/main.go

```diff
@@ -26,29 +26,26 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+// checkConfig will ensure that the required fields are set
+// in the config. This will also check to ensure that the socket
+// path exists.
 func checkConfig(cfg firecracker.Config) error {
```

Comment:
I think it'll be useful moving `Config` validations to `firecracker` package.

Reply:
Cool, moved to `firecracker` package.

Reply:
Can you also validate configuration passed to `NewMachine`? To make sure input is correct.

Reply:
Hm, do we want want to validate in there? I think users could call `Validate` themselves on the config to ensure their configuration file is correct, similarly to how it is being done in `firectl`. That way validating is optional. If `firecracker` drops a required field, users could turn off validation until the SDK has been updated. What do you think? I can be swayed either way.

Reply:
I think we need validation at some point. It can be either `NewMachine` or `Init`.
Let's say user forgot to pass `KernelImagePath`, we want to fail as early as possible and not run Firecracker process at all. Also user might forget to call `Validate`, which will add more confusion than flexibility. Doing that in `NewMachine` feels more 'go way' to me. 

@nmeyerhans do you have opinion regarding this?